src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class Parser {
+
+    private TaskList taskList;
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+    String parse(String input) {
+        if (input.equals("bye")) {
+            return sayBye();
+        } else if (input.equals("list")) {
+            return returnList(taskList);
+        } else if (input.startsWith("mark")) {
+            return mark(input);
+        } else if (input.startsWith("unmark")) {
+            return unmark(input);
+        } else if (input.startsWith("delete")) {
+            return delete(input);
+        } else if (input.startsWith("find")) {
+            return find(input);
+        } else {
+            return addTask(input)[1];
+        }
+    }
+    private String returnList(List<Task> taskList) {
+        StringBuilder out = new StringBuilder();
+        for (int i = 1; i <= taskList.size(); i++) {
+            out.append(String.format("%d. %s\n", i, taskList.get(i - 1)));
+        }
+        return out.toString();
+    }
+
+    String mark(String input) {
+        int i = Integer.parseInt(input.split(" ", 2)[1]) - 1;
+        Task task = taskList.get(i);
+        task.mark();
+        return String.format("\tNice! I've marked this task as done:\n\t  %s", task);
+    }
+
+    private String unmark(String input) {
+        int i = Integer.parseInt(input.split(" ", 2)[1]) - 1;
+        Task task = taskList.remove(i);
+        task.unmark();
+        return String.format("\tNice! I've marked this task as not done yet:\n\t  %s", task);
+    }
```

Comment:
Not sure if it's a missing access modifier for `mark` so I thought I'd let you know in case.